tests/fuzzing.md

```diff
@@ -0,0 +1,79 @@
+# Fuzz testing
+
+Each parser of the library (JSON, BJData, BSON, CBOR, MessagePack, and UBJSON) can be fuzz tested. Currently,
+[LibFuzzer](https://llvm.org/docs/LibFuzzer.html) and [afl++](https://github.com/AFLplusplus/AFLplusplus) are supported.
+
+## Corpus creation
+
+For most effective fuzzing, a [corpus](https://llvm.org/docs/LibFuzzer.html#corpus) should be provided. A corpus is a
+directory with some simple  input files that cover several features of the parser and is hence a good starting point
+for mutations.
+
+```shell
+TEST_DATA_VERSION=3.1.0
+wget https://github.com/nlohmann/json_test_data/archive/refs/tags/v$TEST_DATA_VERSION.zip
+unzip v$TEST_DATA_VERSION.zip
+rm v$TEST_DATA_VERSION.zip
+for FORMAT in json bjdata bson cbor msgpack ubjson
+do
+  rm -fr mkdir corpus_$FORMAT
+  mkdir corpus_$FORMAT
+  find json_test_data-$TEST_DATA_VERSION -size -5k -name "*.$FORMAT" -exec cp "{}" "corpus_$FORMAT" \;
+done
+rm -fr json_test_data-$TEST_DATA_VERSION
+```
+
+The generated corpus can be used with both LibFuzzer and afl++.
+
+## LibFuzzer
+
+To use LibFuzzer, you need to pass `-fsanitize=fuzzer` as `FUZZER_ENGINE`:
+
+```shell
+make fuzzers FUZZER_ENGINE="-fsanitize=fuzzer"
```

Comment:
If CXX isn't set in the current shell or doesn't point to `clang++` built with `libfuzzer`-enabled runtime, it has to be set here.

Reply:
Done: f92e6d0979190347796329a66b76fd9b16ad22a5